extensions/investspec/investspec.py

```diff
@@ -0,0 +1,358 @@
+    return ', '.join(permissions_strings)
+
+
+def format_options_string_from_dict(options):
+    """Represent a dictionary of option: description pairs as a bulleted list.
+
+    Args:
+        options (dict): the dictionary of options to document, where keys are
+            options and values are descriptions of the options
+
+    Returns:
+        list of RST-formatted strings, where each is a line in a bullet list
+    """
+    lines = []
+    # casefold() is a more aggressive version of lower() that may work better
+    # for some languages to remove all case distinctions
+    sorted_options = sorted(
+        list(options.keys()),
+        key=lambda option: option.casefold()
+    )
+    for option in sorted_options:
+        lines.append(f'- {option}: {options[option]}')
+    return lines
+
+
+def format_options_string_from_list(options):
+    """Represent options as a comma-separated list.
+
+    Args:
+        options (list[str]): the set of options to document
+
+    Returns:
+        string of comma-separated options
+    """
+    return ', '.join(options)
+
+
+def capitalize(name):
+
+    def capitalize_word(word):
+        if word in {'of', 'the'}:
+            return word
+        else:
+            return word[0].upper() + word[1:]
+
+    name = ' '.join([capitalize_word(word) for word in name.split(' ')])
+    name = '/'.join([capitalize_word(word) for word in name.split('/')])
+    return name
+
+
+def format_arg(name, spec):
+    """Format an arg spec into user-friendly documentation.
+
+    This is used for documenting:
+        - a single top-level arg
+        - a row or column in a CSV
+        - a field in a vector
+        - an item in a directory
+
+    Args:
+        name (str): Name to give the section. For top-level args this is
+            arg['name']. For nested args it's typically their key in the
+            dictionary one level up.
+        spec (dict): A arg spec dictionary that conforms to the InVEST args
+            spec specification. It must at least have the key `'type'`, and
+            whatever other keys are expected for that type.
+    Returns:
+        list of strings, where each string is a line of RST-formatted text.
+        The first line has the arg name, type, required state, description,
+        and units if applicable. Depending on the type, there may be additional
+        lines that are indented, that describe details of the arg such as
+        vector fields and geometries, option_string options, etc.
+    """
+    type_string = format_type_string(spec['type'])
+    in_parentheses = [type_string]
+
+    # Represent the required state as a string, defaulting to required
+    # It doesn't make sense to include this for boolean checkboxes
+    if spec['type'] != 'boolean':
+        # get() returns None if the key doesn't exist in the dictionary
+        required_string = format_required_string(spec.get('required'))
+        in_parentheses.append(required_string)
```

Comment:
That sounds good to me. I modified it so that the units go right after `number` or `raster`, the units are bolded, and then _required_ is always last:
> (number, units: **m**, _required_)